packages/web-components/@sfx/search-box/src/search-box.ts

```diff
@@ -0,0 +1,205 @@
+      detail: this.value,
+      bubbles: true
+    });
+    this.dispatchEvent(searchboxRequestEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the input box has been cleared.
+   * Invoked in response to a click on the clear button.
+   */
+  emitSearchBoxClearClick() {
+    const searchboxClearedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLEAR_CLICK, { bubbles: true });
+    this.dispatchEvent(searchboxClearedEvent);
+  }
+
+  /**
+   * Updates the contents of the search input box and the value property
+   * with the payload of the given event.
+   * Invoked in response to an update search term event.
+   *
+   * @param e The event object.
+   */
+  updateText(e: CustomEvent) {
+    this.getInputElement().value = e.detail;
+    this.updateSearchTermValue(e.detail)
+  }
+
+  /**
+   * Returns the searchbox input element
+   */
+  getInputElement(): HTMLInputElement {
+    return this.querySelector('[data-sfx-ref=searchInput]') as HTMLInputElement;
+  }
+
+  /**
+   * Dispatches an `sfx::on_search_box_change` event, and updates the value
+   * property with the payload of the given event.
+   *
+   * Invoked in response to a change to the contents of the searchbox.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleChange(e: KeyboardEvent) {
+    this.updateSearchTermValue((e.target as HTMLInputElement).value);
+    this.dispatchEvent(
+      new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CHANGE, {
+        detail: (e.target as HTMLInputElement).value,
+        bubbles: true
+      })
+    );
+  }
+
+  /**
+   * Invokes the emitSearchEvent function when a user presses enter and 
+   * the value property length is greater than 0.
+   *
+   * @param e The KeyboardEvent object.
+   */
+  handleKeydown(e: KeyboardEvent) {
+    if (e.keyCode === KEY_CODES.ENTER && this.value.length > 0) {
+      this.emitSearchEvent();
+    }
+  }
+
+  /**
+   * Updates the value property to the value passed to it.
+   *
+   * @param term The value pulled directly from the input box.
+   */
+  updateSearchTermValue(term: string) {
+    this.value = term;
+  }
+
+  /**
+   * Sets value property and searchbox input value to an empty string and
+   * invokes the emitSearchBoxClearedEvent function.
+   * Invoked in response to click on `clear` button.
+   */
+  clearSearch() {
+    this.getInputElement().value = '';
+    this.value = '';
+    this.emitSearchBoxClearClick();
+  }
+
+  /**
+   * Dispatches an event notifying that the search box input bar has been clicked.
+   * Invoked in response to a user clicking inside of the searchbox input.
+   */
+  clickExposed() {
+    const searchBoxClickedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLICK, { bubbles: true });
+    this.dispatchEvent(searchBoxClickedEvent);
+  }
+
+  /**
+   * Dispatches an event notifying that the search box input bar has been hovered.
+   * Invoked in response to a user hovering inside of the searchbox input.
+   */
+  hoverExposed() {
```

Comment:
What would the use case be for listening for this event? If we can't think of anything, perhaps we'll leave it out for now and put it in if someone asks for it later.